pkg/js/util/exceptions.ts

```diff
@@ -0,0 +1,306 @@
+import { ModelValidationSingleError, ValidationError } from "../errors";
+import { Keyword, ReservedKeywords } from "../validator/Keywords";
+
+interface ValidationErrorProps {
+    message: string,
+    lines: string[],
+    lineIndex: number,
+    metadata: {
+        symbol: string,
+        errorType: ValidationError,
+        relation?: string,
+        typeName?: string
+    }
+    customResolver?: (wordIndex: number, rawLine: string, symbol: string) => number,
+}
+
+interface BaseProps {
+    errors: ModelValidationSingleError[],
+    lines: string[],
+    lineIndex: number,
+    symbol: string,
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getValidRelationsArray = (validRelations?: any, typeName?: string): string[] => {
```

Comment:
Can you set the type of `validRelations` instead of any?

I'm guessing something along the lines of: `Record<string, Record<string, bool>> | string[]`

Reply:
Even better, can delete this. Its a relic from `syntax-tranformer` It always returns `[]` or `undefined` for the error I was trying to validate. 

Even in `sytax-transformer` it had the same behavior.

Removed the `IsValid` check and all tests continue passing.